sale_block_no_stock/wizard/sale_order_block_wizard.py

```diff
@@ -0,0 +1,217 @@
+from odoo import _, api, exceptions, fields, models
+from odoo.tools import groupby
+
+
+class SaleOrderBlockWizard(models.TransientModel):
+    _name = "sale.order.block.wizard"
+    _description = "Sale Order Block Wizard"
+    _transient_max_hours = 0.25  # 15 minutes until destroyed
+
+    sale_line_block_ids = fields.One2many(
+        comodel_name="sale.order.block.wizard.line",
+        inverse_name="wizard_id",
+        string="Sale Block Lines",
+    )
+    confirmation_allowed = fields.Boolean(
+        string="Allowed to confirm",
+        compute="_compute_confirmation_allowed",
+    )
+
+    @api.depends_context("uid")
+    @api.depends("sale_line_block_ids.order_id.company_id")
+    def _compute_confirmation_allowed(self):
+        self.confirmation_allowed = False
+        for record in self:
+            if not record.sale_line_block_ids:
+                continue
+            # All lines must belong to the same company, so choose first one
+            company = record.sale_line_block_ids[0].order_id.company_id
+            record.confirmation_allowed = any(
+                self.env.user in group.users
+                for group in company.sale_line_block_allowed_groups
+            )
+
+    def confirm(self):
+        if not all(self.mapped("confirmation_allowed")):
+            raise exceptions.UserError(
+                _("You are not allowed to confirm these orders.")
+            )
+        orders = self.mapped("sale_line_block_ids.order_id")
+        orders.message_post(
+            body=_("Order confirmed with errors by %s.", self.env.user.name)
```

Comment:
issue: shouldn't you indicate somehow that the message posted is an internal note and not a public message? I don't think customers like to get an email with this text...

Reply:
Default subtype_id is notifications, but addressed.